src/components/schedule-visualizer/robots-overlay.tsx

```diff
@@ -1,25 +1,41 @@
 import * as RomiCore from '@osrf/romi-js-core-interfaces';
 import * as L from 'leaflet';
-import React from 'react';
+import React, { useMemo } from 'react';
 import ColorManager from './colors';
 import Robot from './robot';
 import SVGOverlay, { SVGOverlayProps } from './svg-overlay';
 
 export interface RobotsOverlayProps extends SVGOverlayProps {
   robots: readonly RomiCore.RobotState[];
   colorManager: ColorManager;
+  fleets: readonly RomiCore.FleetState[];
```

Comment:
Since we are receiving the list of fleets here, we might not need `robots` anymore, it should contain the list of robots already.

Reply:
Yeah, we can do that if `RobotsOverlayProps` receives the `currentLevel` too. With the current level, we will be able to get the robots corresponding to that level. Which I think is a good idea, to encapsulate the logic of the robot on the overlay, also we'll be removing some of the robot logic of the ScheduleVisualizer component, which is getting bigger and bigger.

Reply:
That can work. Currently it only receives a list of robots you that it only receives what it is going to render, receiving all robots would mean we need bring that logic in, there is a bit of refactoring but there shouldn't be any problems with that.